src/charm.py

```diff
@@ -78,96 +53,53 @@ def __init__(self, *args):
-                    stop=stop_after_delay(max_delay=self._max_time_checking_resources),
-                    wait=wait_exponential(multiplier=0.1, min=0.1, max=15),
-                    reraise=True,
-                ):
-                    with attempt:
-                        self.logger.info(
-                            f"Checking status of requested resources (attempt "
-                            f"{attempt.retry_state.attempt_number})"
-                        )
-                        self._check_deployed_resources()
-            except CheckFailedError:
-                self.unit.status = BlockedStatus(
-                    "Some Kubernetes resources did not start correctly during install"
-                )
-                return
-
-            # Otherwise, application is working as expected
-            self.model.unit.status = ActiveStatus()
-
-    @only_leader
-    def remove(self, event):
-        """Remove charm."""
-
-        self.model.unit.status = MaintenanceStatus("Calculating manifests")
-        self.ensure_state()
-
-        manifest = self.get_manifest()
-
-        self.model.unit.status = MaintenanceStatus("Removing manifests")
-
-        self.remove_manifest(manifest)
-
-    def ensure_state(self):
-        self.state.set_default(
-            username="admin",
-            password="".join(choices(ascii_letters, k=30)),
-            salt=bcrypt.gensalt(),
-            user_id=str(uuid4()),
-        )
-
-    def get_manifest(self):
-        # Handle ingress
-        ingress = self._get_interface("ingress")
-
-        if ingress:
-            for app_name, version in ingress.versions.items():
-                data = {
-                    "prefix": "/dex",
-                    "rewrite": "/dex",
-                    "service": self.model.app.name,
-                    "port": self.model.config["port"],
+    @property
+    def _dex_auth_layer(self) -> Layer:
+        """Returns a pre-configured Pebble layer."""
+
+        layer_config = {
+            "summary": "dex-auth-operator layer",
+            "description": "pebble config layer for dex-auth-operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "entrypoint of the dex-auth-operator image",
+                    "command": f"{self._entrypoint} dex serve {self._dex_config_path}",
+                    "startup": "enabled",
+                    "environment": {
+                        "KUBERNETES_POD_NAMESPACE": self._namespace,
+                    },
                 }
+            },
+        }
+        return Layer(layer_config)
 
-                ingress.send_data(data, app_name)
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+        if not self._container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for pod startup to complete")
+            return
```

Comment:
When `not can_connect()`, we set the `WaitingStatus` but then return normally.  That means that `main` will continue running and (I think always) overwrite this status with something else, possibly `Active`, even through the container is not up.  Is that the intent?  

My guess is that we want to actually break out of the event here (raise an exception, main catches it/sets status, and main returns) rather than just set the status, but I might be missing a nuance.